tests/test_graphBuilder.py

```diff
@@ -25,17 +25,23 @@
 
 from __future__ import absolute_import, division, print_function
 
+import os
 import unittest
-from builtins import object
 
 import lsst.utils.tests
+from lsst.daf.butler.core.registry import Registry
 import lsst.pex.config as pexConfig
 import lsst.pipe.base as pipeBase
 from lsst.pipe.supertask import (GraphBuilder, Pipeline, SuperTask, TaskDef,
                                  SuperTaskConfig, InputDatasetConfig,
                                  OutputDatasetConfig)
+from lsst.pipe.supertask.graphBuilder import _TaskDatasetTypes
 from lsst.pipe.supertask.examples.exampleStorageClass import ExampleStorageClass  # noqa: F401
 
+# config files for registry
+_registry_config = os.path.join(os.environ["DAF_BUTLER_DIR"], "config", "registry.yaml")
+_schema_config = os.path.join(os.environ["DAF_BUTLER_DIR"], "config", "schema.yaml")
```

Comment:
I think you should be able to instantiate an empty `SchemaConfig` and `RegistryConfig` and they should be populated from the above defaults. You shouldn't have to know where the butler default configs are stored. You can give these config instances to `fromConfig()` below.

Reply:
I have changed it to:

    reg = Registry.fromConfig(RegistryConfig())

and it seems to work OK. If this is a reasonable/useful default should not first argument be optional too?

Reply:
I guess `ConfigSubset` reads defaults from known location so `SchemaConfig` is initialized with defaults from a file and and then optionally updated from other config object?

Reply:
Looking at the code further I'm not sure what happens when `SchemaConfig` is `update()`d from `RegistryConfig`. I guess I do need to pass `SchemaConfig` instance.

Reply:
`SchemaConfig` doesn't have registry information in it. `ButlerConfig` will have both schema and registry defaults. In answer to the other question, you can provide a config file and it will be merged with all defaults read from `$DAF_BUTLER_CONFIG_PATH` and `$DAF_BUTLER_DIR/config`